bin/proving-service/src/proxy/worker.rs

```diff
@@ -71,6 +88,35 @@ impl Worker {
     pub fn set_availability(&mut self, is_available: bool) {
         self.is_available = is_available;
     }
+
+    pub fn mark_as_unhealthy(&mut self) {
```

Comment:
nit: Would be good to get docs on this and the `has_exceeded_failed_health_checks` function. Also, maybe these mutator functions should be grouped together.